Modules/_posixsubprocess.c

```diff
@@ -767,8 +767,10 @@ child_exec(char *const exec_array[],
 #endif
 
 #ifdef HAVE_SETGROUPS
-    if (extra_group_size > 0)
+    if (extra_group_size >= 0) {
+        assert((extra_group_size == 0 && extra_groups == NULL) || extra_group_size);
```

Comment:
Should not extra_groups be not NULL if extra_group_size is not 0 (and not negative)?
```suggestion
        assert((extra_group_size == 0) == (extra_groups == NULL));
```

Reply:
thanks, that's a nicer assertion.